packages/server/graphql/mutations/addIntegrationProvider.ts

```diff
@@ -35,22 +36,43 @@ const addIntegrationProvider = {
     const subOptions = {mutatorId, operationId}
 
     // AUTH
-    if (!isTeamMember(authToken, teamId)) {
+    if (!isTeamMember(authToken, teamId) && !isSuperUser(authToken)) {
```

Comment:
-1 this can't be a superUser because we grab the viewerId from L33 and provide it in the data payload. that means the data payload will return the superuser to all members of the team & we may not want that team to have that info on the superuser

Reply:
The use case for this is us adding the integration provider on behalf of the organization. Given that the viewerId is only used for the subscription which has a big TODO in line 80 I would argue that the AddIntegrationProviderPayload needs rework. I think it should only have `provider` and `teamId`/`team` fields. It should also be sent to all teams in the org for org scope. The reason is that the `IntegrationProvider` itself is not bound to a `User` or `TeamMember`, but should be available to all TeamMembers in `sharedProviders` field.

Reply:
+1 oh! that use case makes sense, i didn't consider that.
removing the user field in the payload definitely works.
alternatively, we could get the userId from the authToken.sub and that way each subscriber would only get _their_ TeamMember object